### PR TITLE
Refactor MLEngine code and add deploy and set_default commands

### PR DESCRIPTION
**Pull request by hongye-sun**
The PR includes changes:

* Add deploy and set_default commands
* Refactor the model and version name to be the full API path to be consistent with the API representation.
* Dump files in /tmp/kfp/output folder.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/864)

### CODE REVIEW

File: component_sdk/python/kfp_component/google/ml_engine/_client.py [hunk 1]
```diff
@@ -81,79 +81,75 @@ def create_model(self, project_id, model):
...
         Args:
-            project_id: the ID of the parent project.
             model_name: the name of the parent model.
             version: the payload of the version.
 
         Returns:
             The created version.
         """
         return self._ml_client.projects().models().versions().create(
-            parent = 'projects/{}/models/{}'.format(project_id, model_name),
```
Comment (gaoning777) [inline component_sdk/python/kfp_component/google/ml_engine/_client.py]: what is the default project id?

Comment (hongye-sun) [reply component_sdk/python/kfp_component/google/ml_engine/_client.py]: There is no default project id. It's a required field that user has to pass data in.

Comment (gaoning777) [reply component_sdk/python/kfp_component/google/ml_engine/_client.py]: never mind. I read the restful api doc.

File: component_sdk/python/kfp_component/google/ml_engine/_create_version.py [hunk 2]
```diff
@@ -60,16 +60,17 @@ def create_version(project_id, model_name, deployemnt_uri=None, version_name=Non
...
 class CreateVersionOp:
-    def __init__(self, project_id, model_name, version, 
+    def __init__(self, model_name, version, 
         replace_existing, wait_interval):
         self._ml = MLEngineClient()
-        self._project_id = project_id
-        self._model_name = gcp_common.normalize_name(model_name)
+        self._model_name = model_name
+        self._project_id, self._model_short_name = self._parse_model_name(model_name)
         self._version_name = None
```
Comment (gaoning777) [inline component_sdk/python/kfp_component/google/ml_engine/_create_version.py]: Could we add a comment to illustrate the difference among _version_name, _version_shor_name, and _version

File: component_sdk/python/kfp_component/google/ml_engine/_create_version.py [hunk 3]
```diff
@@ -25,13 +26,12 @@
 from ._common_ops import wait_existing_version, wait_for_operation_done
 
 @decorators.SetParseFns(python_version=str, runtime_version=str)
-def create_version(project_id, model_name, deployemnt_uri=None, version_name=None,
```
Comment (gaoning777) [inline component_sdk/python/kfp_component/google/ml_engine/_create_version.py]: Why expose the model_name and parse the model name? how about keep the project_id and construct the model_name inside the op?

File: component_sdk/python/kfp_component/google/ml_engine/_delete_version.py [hunk 4]
```diff
@@ -22,39 +22,33 @@
...
-        model_name (str): required, the name of the parent model.
         version_name (str): required, the name of the version.
         wait_interval (int): the interval to wait for a long running operation.
     """
-    DeleteVersionOp(project_id, model_name, version_name, 
-        wait_interval).execute_and_wait()
+    DeleteVersionOp(version_name, wait_interval).execute_and_wait()
 
 class DeleteVersionOp:
-    def __init__(self, project_id, model_name, version_name, wait_interval):
```
Comment (gaoning777) [inline component_sdk/python/kfp_component/google/ml_engine/_delete_version.py]: Same as above, why not expose the project_id, model_name to construct the version_name?

File: component_sdk/python/kfp_component/google/ml_engine/_create_version.py (hunk 2 above)
Comment (hongye-sun) [reply component_sdk/python/kfp_component/google/ml_engine/_create_version.py]: done

File: component_sdk/python/kfp_component/google/ml_engine/_create_version.py (hunk 3 above)
Comment (hongye-sun) [reply component_sdk/python/kfp_component/google/ml_engine/_create_version.py]: Discussed offline. Rename short names to be name or model_short_name and changed comment to be reflect that.

File: component_sdk/python/kfp_component/google/ml_engine/_delete_version.py (hunk 4 above)
Comment (hongye-sun) [reply component_sdk/python/kfp_component/google/ml_engine/_delete_version.py]: ditto

File: component_sdk/python/kfp_component/google/ml_engine/_create_version.py (hunk 3 above)
Comment (gaoning777) [reply component_sdk/python/kfp_component/google/ml_engine/_create_version.py]: Discussed offline added. change the user provided name to id, for example model_id, project_id, version_id. The returned is model_name, version_name.

**Comment by hongye-sun** [comment]
Thanks for the review and great comments.

**Pull request merged**